media/com_cache/js/admin-cache-default.es6.js

```diff
@@ -0,0 +1,18 @@
+/**
+ * @copyright   Copyright (C) 2005 - 2017 Open Source Matters, Inc. All rights reserved.
+ * @license     GNU General Public License version 2 or later; see LICENSE.txt
+ */
+
+Joomla = window.Joomla || {};
+
+((document, Joomla) => {
+  'use strict';
+
+  document.addEventListener('DOMContentLoaded', () => {
+    [].slice.call(document.querySelectorAll('.cache-entry')).forEach((el) => {
+      el.addEventListener('click', () => {
+        Joomla.isChecked(this.checked);
```

Comment:
this in this case is the window object and not the element (arrow function).
You can use Joomla.isChecked(event.currentTarget.checked)

Reply:
Ok. Why does it work then? Is this code needed at all?

Reply:
For me it does not work as it should. 
`<input type="hidden" name="boxchecked">` is wrongly decremented with every click. 

boxchecked count is needed for the toolbar button when we check if any item is selected before deletion. 

reproduce:

- create folder in cache directory
- open cache manager in backend
- check item
- uncheck it
- click delete 
- Dont see "Please first make a selection from the list."

Reply:
@dneukirchen If I'm not mistaken, `this` is the same as `event.currentTarget`

Reply:
Not in arrow functions. The value of `this` inside the arrow function is determined by where the arrow function is defined, not where it is used (the element)

Reply:
This explains the strange code on the top of the normal js file with _this. Fixed.